From 86e56a6d493feba566f51269266197ed820bac98 Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Wed, 30 Nov 2016 10:40:29 +0100
Subject: [PATCH 144/206] Initialize focus manager in onResume().

Since commit 87b4a40d588a7ae8a61cb353c4cff8df37f7b483 moved camera
initialization (and focus manager initialization with it) to a
background thread, the focus manager initialization now races against
the onPreviewRectChanged() call triggered by SurfaceView initialization
in the main thread, potentially leaving the focus manager in an
uninitialized state that prevented taking pictures.

Change-Id: I8cf650d6a67768acd131b3cddad175ed198e0838
---
 src/com/android/camera/PhotoModule.java | 1 +
 src/com/android/camera/VideoModule.java | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index e7436ed4b..a393467e3 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -2691,6 +2691,7 @@ public class PhotoModule
     @Override
     public void onResumeBeforeSuper() {
         mPaused = false;
+        if (mFocusManager == null) initializeFocusManager();
     }
 
     private void openCamera() {
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index bc3a6417e..9e629712f 100755
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1231,6 +1231,7 @@ public class VideoModule implements CameraModule,
     @Override
     public void onResumeBeforeSuper() {
         mPaused = false;
+        if (mFocusManager == null) initializeFocusManager();
     }
 
     @Override
-- 
2.17.1

